src/main/java/org/terasology/core/world/generator/rasterizers/FloraRasterizer.java

```diff
@@ -69,40 +67,18 @@ public void initialize() {
     @Override
     public void generateChunk(CoreChunk chunk, Region chunkRegion) {
         FloraFacet facet = chunkRegion.getFacet(FloraFacet.class);
-        WhiteNoise noise = new WhiteNoise(chunk.getPosition().hashCode());
 
-        facet.getRelativeEntries().entrySet().stream()
-                .filter(isPlacementAllowed(chunk))
-                .forEach(setBlock(chunk, noise));
-    }
-
-    /**
-     * A predicate that checks whether - some other rasterizer has already placed something here - block is "supported"
-     * by block below
-     */
-    private Predicate<Map.Entry<BaseVector3i, FloraType>> isPlacementAllowed(CoreChunk chunk) {
-        return entry -> {
-            BaseVector3i pos = entry.getKey();
-            return hasSupport(chunk, pos) && isFree(chunk, pos);
-        };
-    }
-
-    private boolean isFree(CoreChunk chunk, BaseVector3i pos) {
-        return chunk.getBlock(pos).equals(air);
-    }
+        WhiteNoise noise = new WhiteNoise(chunk.getPosition().hashCode());
 
-    private boolean hasSupport(CoreChunk chunk, BaseVector3i pos) {
```

Comment:
So we can get rid of this check now because flora will only be placed in locations where it makes sense (based on the updated facets), right?

Reply:
Basically, yes. Wherever this check would have removed a plant previously, that case should now be dealt with by changing the SurfacesFacet to mark that place as not a surface, like I've done in the Caves PR. This will cause the floating plant problem to temporarily reappear with Lakes until I update that module too, but I thought it was better to make the actual problem that needs to be fixed (incorrect values of the SurfacesFacet) more obvious, rather than hiding it.